pkg/controllers/managedos/managedos.go

```diff
@@ -93,12 +93,12 @@ func (h *handler) OnChange(mos *elm.ManagedOSImage, status elm.ManagedOSImageSta
 
 	bundle := &v1alpha1.Bundle{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      name.SafeConcatName("mos", mos.Name),
+			Name:      name.SafeConcatName("mos", mos.Name, string(mos.UID)),
 			Namespace: mos.Namespace,
 		},
 		Spec: v1alpha1.BundleSpec{
 			Resources:               resources,
-			BundleDeploymentOptions: v1alpha1.BundleDeploymentOptions{},
+			BundleDeploymentOptions: v1alpha1.BundleDeploymentOptions{DefaultNamespace: mos.Namespace},
```

Comment:
This missing meant that we were deploying the bundle in the "default" namespace which is not bad, but also not good. Should be deployed on the same namespace :D